usecases/account.go

```diff
@@ -13,24 +38,98 @@ type AccountUseCasesInterface interface {
 }
 
 type AccountUseCases struct {
+	AccountStorage accountstorage.Interface
+	Auth           auth.Interface
 }
 
-func (AccountUseCases) CreateAccount(login string, password string) (Account, error) {
-	panic("not implemented method")
+func (a* AccountUseCases) CreateAccount(login string, password string) (Account, error) {
+	if err := validateLogin(login); err != nil {
+		return Account{}, err
+	}
+	if err := validatePassword(password); err != nil {
+		return Account{}, err
+	}
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	if err != nil {
+		return Account{}, err
+	}
+	acc, err := a.AccountStorage.CreateAccount(accountstorage.Credentials{
+		Login:    login,
+		Password: string(hashedPassword),
+	})
+	if err != nil {
+		return Account{}, err
+	}
+	return Account{Id: acc.Id}, nil
 }
 
-func (AccountUseCases) GetAccountById(id string) (Account, error) {
-	panic("not implemented method")
+func (a* AccountUseCases) GetAccountById(id string) (Account, error) {
+	acc, err := a.AccountStorage.GetAccountById(id)
+	if err != nil {
+		return Account{}, err
+	}
+	return Account{Id: acc.Id}, err
 }
 
-func (AccountUseCases) LoginToAccount(login string, password string) (string, error) {
-	panic("not implemented method")
+func (a* AccountUseCases) LoginToAccount(login string, password string) (string, error) {
+	if err := validateLogin(login); err != nil {
+		return "", err
+	}
+	if err := validatePassword(password); err != nil {
+		return "", err
+	}
+	acc, err := a.AccountStorage.GetAccountByLogin(login)
+	if err != nil {
+		return "", err
+	}
+	if err := bcrypt.CompareHashAndPassword([]byte(acc.Credentials.Password), []byte(password)); err != nil {
+		return "", err
+	}
+	token, err := a.Auth.IssueToken(acc.Id)
+	if err != nil {
+		return "", err
+	}
+	return token, err
 }
 
-func (AccountUseCases) Authenticate(token string) (string, error) {
-	panic("not implemented method")
+func (a* AccountUseCases) Authenticate(token string) (string, error) {
+	return a.Auth.UserIdByToken(token)
 }
 
-func (AccountUseCases) Logout() () {
+func (a* AccountUseCases) Logout() () {
 	panic("not implemented method")
+}
+
+func validateLogin(login string) error {
+	chars := 0
+	for _, r := range login {
+		if !unicode.IsLetter(r) && !unicode.IsDigit(r) {
+			return ErrInvalidLoginString
+		}
+		chars++
+	}
+	if chars < minLoginLength {
+		return ErrTooShortLogin
+	}
+	if chars > maxLoginLength {
+		return ErrTooLongLogin
+	}
+	return nil
+}
+
+func validatePassword(password string) error {
+	chars := 0
```

Comment:
Please do me a personal favor, numChars or nChars might be a better var naming. =)